include/seqan3/io/record.hpp

```diff
@@ -246,16 +246,34 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
                   "You must give as many IDs as types to seqan3::record.");
 
     //!\brief (Re-)Initialise all tuple elements with `{}`.
-    void clear()
+    void clear() noexcept(noexcept(clear_impl(*this, std::make_integer_sequence<size_t, field_types::size()>{})))
     {
         clear_impl(*this, std::make_integer_sequence<size_t, field_types::size()>{});
```

Comment:
```suggestion
        std::apply([this] (auto & ...args) { (clear_element(args),...); }, *this);
```
Then with custom detail concept:
```cpp
template <typename t>
SEQAN3_CONCEPT has_clear_member_function = requires (t & e) { {e.clear() } -> void; };
```

and proper private member function
```cpp
template <typename element_t>
void clear_element(element_t & element)
{
    if constexpr (has_clear_member_function<element_t>)
        element.clear();
    else
        element = element_t{};
}
```

Reply:
How am I suppose to adjust the conditional noexcept in this version of a combined `clear_element` ?

Reply:
Why do we need an extra concept if this is only used on one place? I thought the policy was to only externalise these if they are used in multiple places.

I also think that it is not possible to easily define `noexcept` with the `if constexpr`. I like the `std::apply`, though. But in general, maybe we should spend too much time discussed this mini-PR?

Reply:
Well in C++20 we don't need an extra concept as this:
```cpp
    if constexpr (requires (element_t & e) { {e.clear()} -> std::same_as<void>;})
        element.clear();
    else
        element = element_t{};
```
would be valid code. But, well, that is future.